ibis/pandas/core.py

```diff
@@ -243,19 +260,27 @@ def execute_until_in_scope(
     computable_args = [arg for arg in op.inputs if is_computable_input(arg)]
 
     # recursively compute each node's arguments until we've changed type
-    scopes = [
-        execute_until_in_scope(
-            arg,
-            new_scope,
-            aggcontext=aggcontext,
-            post_execute_=post_execute_,
-            clients=clients,
-            **kwargs,
+    if len(computed_localcontexts) == len(computable_args):
+        scopes = [
+            execute_until_in_scope(
+                arg,
+                new_scope,
+                localcontext=localcontext,
+                aggcontext=aggcontext,
+                post_execute_=post_execute_,
+                clients=clients,
+                **kwargs,
+            )
+            if hasattr(arg, 'op')
+            else {arg: arg}
+            for (arg, localcontext) in zip(
+                computable_args, computed_localcontexts
+            )
+        ]
+    else:
+        raise com.IbisError(
+            "computed_localcontexts differ with computable_arg in length."
```

Comment:
It would be helpful to include type(op) inside the error message

Reply:
+1 Done